Fix crash of SobolEngine if default tensor type is cuda

Balandat opened the pull request:
Summary: Addresses https://github.com/pytorch/pytorch/issues/32494

Test Plan:
```
import torch
from torch.quasirandom import SobolEngine

torch.set_default_tensor_type(torch.cuda.FloatTensor)
se = SobolEngine(3)
```

Differential Revision: D19517571

Balandat commented on torch/quasirandom.py:
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
```
Not necessary, but since master is py3 only I figure this is the way to go forward (otherwise linters are confused etc.).

vishwakftw replied:
Other files such as `torch/random.py/` or `torch/storage.py` don't have this, hence the concern.

Balandat replied:
Shouldn’t matter, but sure, I can remove it. See https://github.com/pytorch/pytorch/issues/23795#issuecomment-577765664 for some discussion.

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 63f02dd2:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=6ad9e5c70d1f496b9eaca191ec18e26c106c6436&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base 6ad9e5c7 since Jan 25
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=63f02dd25260fc09d2d50bc0b8cd9aa8fc0c6f12).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=90891927) from Jan 25



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Balandat committed:
Fix crash of SobolEngine if default tensor type is cuda (#32496)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32496

Addresses https://github.com/pytorch/pytorch/issues/32494

Test Plan:
```
import torch
from torch.quasirandom import SobolEngine

torch.set_default_tensor_type(torch.cuda.FloatTensor)
se = SobolEngine(3)
```

Differential Revision: D19517571

fbshipit-source-id: 5b58efaaa30345b0d6f22222c8e98734a9b54858

Balandat commented:
None of the test failures are related. I also removed the py3 shebang. @vishwakftw does this look ok?

vishwakftw reviewed (approved):
Sorry about the delay, this looks good to me! Thank you.
